Changes to bring changes into develop from GFSv16.3.25

ADCollard opened the pull request:
This adds in scripts changes to turn on the CADS cloud detection code [GSI #428](https://github.com/NOAA-EMC/GSI/issues/428) and the `optconv` modification to improve GSI minimisation [GSI #375](https://github.com/NOAA-EMC/GSI/issues/375)

Refs #2987
Resolves #2934 

# Type of change
- [x] New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? YES
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO 

# How has this been tested?
Changes have been tested in individual parallel experiments.


# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] This change is covered by an existing CI test or a new one has been added
- [x] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [x] I have made corresponding changes to the system documentation if necessary

ADCollard committed:
Add CADS and optconv options to be consistent with v16.4

ADCollard committed:
Add missing optconv namelist entry

ADCollard committed:
Merge remote-tracking branch 'upstream/develop' into develop

RussTreadon-NOAA commented:
A check of a `gdas_anal.log` from a WCOSS2 GFS v17 parallel shows 
```
OPTCONV =  0.000000000000000E+000,
...
 AIRS_CADS       = F,
 CRIS_CADS       = F,
 IASI_CADS       = F,
 IASING_CADS     = F
```

When this PR is merged into g-w `develop` the default behavior changes to 
```
OPTCONV =  6.000000000000000E-002,
...
 AIRS_CADS       = T,
 CRIS_CADS       = T,
 IASI_CADS       = T,
 IASING_CADS     = F
```

We should add Cathy as a reviewer or tag her on this PR.

ADCollard commented:
> A check of a `gdas_anal.log` from a WCOSS2 GFS v17 parallel shows
> 
> ```
> OPTCONV =  0.000000000000000E+000,
> ...
>  AIRS_CADS       = F,
>  CRIS_CADS       = F,
>  IASI_CADS       = F,
>  IASING_CADS     = F
> ```
> 
> When this PR is merged into g-w `develop` the default behavior changes to
> 
> ```
> OPTCONV =  6.000000000000000E-002,
> ...
>  AIRS_CADS       = T,
>  CRIS_CADS       = T,
>  IASI_CADS       = T,
>  IASING_CADS     = F
> ```
> 
> We should add Cathy as a reviewer or tag her on this PR.

@RussTreadon-NOAA Oops!  I meant to add @CatherineThomas-NOAA.  I had already given her a heads up.

CatherineThomas-NOAA commented:
Thanks for the ping @ADCollard.  Is there more testing to do on the v17 side or is this ready to be solidly turned on by default?  We still have [this](https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/config/gfs/config.anal#L153) in `config.anal`:
```
export AIRS_CADS=".false."
export CRIS_CADS=".false."
export IASI_CADS=".false."
```

emilyhcliu commented on scripts/exglobal_atmos_analysis.sh:
```diff
@@ -748,7 +752,7 @@ cat > gsiparm.anl << EOF
   write_fv3_incr=${write_fv3_increment},
   nhr_anal=${IAUFHRS},
-  ta2tb=${ta2tb},
+  ta2tb=${ta2tb},optconv=${optconv},
```
Is this one for turning on the convective clouds?

ADCollard replied:
@emilyhcliu This change downweights observations where the humidity Jacobians are large.  @jderber-NOAA found this improved minimisation.

emilyhcliu replied:
I see!  Thanks.

ADCollard commented:
> Thanks for the ping @ADCollard. Is there more testing to do on the v17 side or is this ready to be solidly turned on by default? We still have [this](https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/config/gfs/config.anal#L153) in `config.anal`:
> 
> ```
> export AIRS_CADS=".false."
> export CRIS_CADS=".false."
> export IASI_CADS=".false."
> ```

@CatherineThomas-NOAA I have run parallels in the v17 context and there is positive impact.   I am getting some slides together to show that.

emilyhcliu reviewed (approved):
Looks good.  Approved!

HaixiaLiu-NOAA reviewed (approved):
Changes look good to me. Approved. 
Thanks for informing me of these changes. Look forward to your slides showing the impact.

ADCollard commented:
Upon further review, I am not comfortable with the CADS changes.   Changing back to a draft PR.

KateFriedman-NOAA commented:
@ADCollard Checking in, what is the status of this PR? Thanks! :)

ADCollard commented:
@KateFriedman-NOAA I have a couple of pending changes that I want to bundle with this.  I hope that is OK.

KateFriedman-NOAA commented:
@ADCollard sure!

ADCollard committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA commented:
@ADCollard could you include turning on the GMI in this PR (issue #2934)?

ADCollard commented:
> @ADCollard could you include turning on the GMI in this PR (issue #2934)?

Yes, I am making sure this is consistent with v16.3.25 right now.   I will hold back on GOES-19 CSRs and updating the CRTM version until that version is on all machines.

ADCollard committed:
Update version of GSI

ADCollard commented:
@KateFriedman-NOAA One issue with reinstating GMI is that the bias correction file is corrupted.   You might find that you get complaints that GSI fails when people run this.   We need to fix this and after v16.3.25 goes in the operational files should be correct.  So I am just giving you a heads up.

Copying @CatherineThomas-NOAA @JessicaMeixner-NOAA for awareness

ADCollard committed:
Add Windborne, Saildrone, GMI and G19 CSRs

KateFriedman-NOAA commented:
> @KateFriedman-NOAA One issue with reinstating GMI is that the bias correction file is corrupted. You might find that you get complaints that GSI fails when people run this. We need to fix this and after v16.3.25 goes in the operational files should be correct. So I am just giving you a heads up.
> 
> Copying @CatherineThomas-NOAA @JessicaMeixner-NOAA for awareness

@ADCollard should we wait to reinstate GMI then? We only have the GMI in the dump archive for the last year (maybe two), is that an issue as well?

ADCollard committed:
Add G19 in a couple of other spots

ADCollard commented:
> > @KateFriedman-NOAA One issue with reinstating GMI is that the bias correction file is corrupted. You might find that you get complaints that GSI fails when people run this. We need to fix this and after v16.3.25 goes in the operational files should be correct. So I am just giving you a heads up.
> > Copying @CatherineThomas-NOAA @JessicaMeixner-NOAA for awareness
> 
> @ADCollard should we wait to reinstate GMI then? We only have the GMI in the dump archive for the last year (maybe two), is that an issue as well?

@KateFriedman-NOAA I think we should just bite the bullet.   The v17 parallels will reset the bias corrections anyway.  Just be prepared to tell people to either comment out GMI or zero the BCs.   Missing data is not an issue.

ADCollard commented:
I just updated the GSI version and made a small number of updates to the scripts to make it consistent with v16.3.25.   

Moved out of draft mode.

ADCollard committed:
Fix an issues picked up by lint-shell (not mine)

KateFriedman-NOAA commented:
> @KateFriedman-NOAA I think we should just bite the bullet. The v17 parallels will reset the bias corrections anyway. Just be prepared to tell people to either comment out GMI or zero the BCs. Missing data is not an issue.

@ADCollard Works for me! So for my knowledge and for letting users know, which scenarios will fail with GMI on? To my knowledge, users can currently only run GFSv17 with our generated staged ICs and ICs from v17 parallels. Which sources will fail?

ADCollard commented:
> > @KateFriedman-NOAA I think we should just bite the bullet. The v17 parallels will reset the bias corrections anyway. Just be prepared to tell people to either comment out GMI or zero the BCs. Missing data is not an issue.
> 
> @ADCollard Works for me! So for my knowledge and for letting users know, which scenarios will fail with GMI on? To my knowledge, users can currently only run GFSv17 with our generated staged ICs and ICs from v17 parallels. Which sources will fail?

@KateFriedman-NOAA The minimisation will fail in GSI.

ADCollard committed:
Merge branch 'develop' into develop

ADCollard commented:
@KateFriedman-NOAA Is there anything else we need to do to push this forward?

KateFriedman-NOAA reviewed (dismissed):
Updates look good, thanks @ADCollard !

KateFriedman-NOAA commented:
@ADCollard please sync your PR branch with `develop`, thanks! I have asked reviewers to sign off again. Once this gets another approval or two I'll run this PR through CI ahead of merging.

CatherineThomas-NOAA reviewed (dismissed):
Looks good to me, thanks @ADCollard!

ADCollard committed:
Merge branch 'NOAA-EMC:develop' into develop

HaixiaLiu-NOAA commented on scripts/exgdas_enkf_update.sh:
```diff
@@ -364,6 +364,7 @@ cat > enkf.nml << EOFnml
    sattypes_rad(77)= 'viirs-m_j2',    dsis(77)= 'viirs-m_j2',
    sattypes_rad(78)= 'atms_n21',      dsis(78)= 'atms_n21',
    sattypes_rad(79)= 'cris-fsr_n21',  dsis(79)= 'cris-fsr_n21',
+   sattypes_rad(80)= 'abi_g19',       dsis(75)= 'abi_g19',
```
typo in this line: dsis for abi_g19 should be the 80th entry rather than the 75th

ADCollard committed:
Update exgdas_enkf_update.sh

Fix typo

ADCollard committed:
Merge branch 'NOAA-EMC:develop' into develop

HaixiaLiu-NOAA reviewed (approved):
all good. approved

KateFriedman-NOAA commented:
CI Tests set up to run in /lfs/h2/emc/ptmp/emc.global/PR/PR_3498/RUNTESTS on WCOSS

CatherineThomas-NOAA commented on scripts/exglobal_atmos_analysis.sh:
```diff
@@ -494,6 +498,8 @@ ${NLN} ${AHIBF}            ahibufr
 ${NLN} ${ABIBF}            abibufr
 ${NLN} ${HDOB}             hdobbufr
 ${NLN} ${SSTVIIRS}         sstviirs
+${NLN} ${SAILDRONE}        sdbufr
```
@ADCollard : Do $SAILDRONE and $GSBBF definitions need to be added to around Line 122?

ADCollard replied:
@CatherineThomas-NOAA Ugg!  How did that get missed.   I will fix.

ADCollard replied:
@CatherineThomas-NOAA Fixed now.

emilyhcliu replied:
> @ADCollard : Do $SAILDRONE and $GSBBF definitions need to be added to around Line 122?

@CatherineThomas-NOAA Nice catch!

ADCollard committed:
Add missing references to SAILDRONE and GSBBF

KateFriedman-NOAA commented:
Thanks @ADCollard ! I pulled the fix into the clone on WCOSS2 and rewound the failed analysis jobs.
```
WCOSS2 (BACKUPSYS-D) global-workflow> git pull
remote: Enumerating objects: 4, done.
remote: Counting objects: 100% (4/4), done.
remote: Total 4 (delta 3), reused 4 (delta 3), pack-reused 0 (from 0)
Unpacking objects: 100% (4/4), 426 bytes | 2.00 KiB/s, done.
From github.com:ADCollard/global-workflow
 * branch                develop    -> FETCH_HEAD
Updating ed90bcfe8..9e367f786
Fast-forward
 scripts/exglobal_atmos_analysis.sh | 2 ++
 1 file changed, 2 insertions(+)
```

KateFriedman-NOAA commented:
Pulled fix into CI clone on Gaea C6 and rewound DEAD jobs.

KateFriedman-NOAA commented:
@ADCollard there is a DEAD job in the `C48mx500_hybAOWCDA` CI test on both WCOSS2 and C6. Looks like additional testing and fixing is needed. Please see this log on WCOSS2: `/lfs/h2/emc/ptmp/emc.global/PR/PR_3498/RUNTESTS/COMROOT/C48mx500_hybAOWCDA_3498/2021032500/gdas_anal.log`

I am suspending CI testing on this PR. Please sync your branch with `develop` again and then test your branch. I suggest running the affected test via `generate_workflows.sh`. Please see this slide from a presentation that @DavidHuber-NOAA gave earlier this week on this new utility: https://docs.google.com/presentation/d/1azLQfbaVmWojkyUB0n73hkbQbN07LKF4/edit?slide=id.p5#slide=id.p5. You can use it to run just the CI test case in question or run all CI tests that are enabled on whichever machine you run on. Let me know if you have questions about using that utility. Thanks!

RussTreadon-NOAA commented:
Looked at `/lfs/h2/emc/ptmp/emc.global/PR/PR_3498/RUNTESTS/COMROOT/C48mx500_hybAOWCDA_3498/logs/2021032500/gdas_anal.log`

See that the job was killed for exceeding the specified `walltime=01:20:00`. 
```
nid001039.dogwood.wcoss2.ncep.noaa.gov 8:  READ_SATWND,nread,ndata,nreal,nodata=     3070150     1312856          27
     2625712
nid001039.dogwood.wcoss2.ncep.noaa.gov 8:
=>> PBS: job killed: walltime 4837 exceeded limit 4800
nid001065.dogwood.wcoss2.ncep.noaa.gov 61: forrtl: error (78): process killed (SIGTERM)
Image              PC                Routine            Line        Source
gsi.x              0000000002087D8B  Unknown               Unknown  Unknown
libpthread-2.31.s  000014AA4AE32910  Unknown               Unknown  Unknown
libmpi_intel.so.1  000014AA3C8FC6D4  Unknown               Unknown  Unknown
libmpi_intel.so.1  000014AA3D4A5A96  Unknown               Unknown  Unknown
libmpi_intel.so.1  000014AA3D4A9F99  Unknown               Unknown  Unknown
libmpi_intel.so.1  000014AA3D62C465  Unknown               Unknown  Unknown
libmpi_intel.so.1  000014AA3D62CB20  Unknown               Unknown  Unknown
libmpi_intel.so.1  000014AA3D3D3C02  Unknown               Unknown  Unknown
libmpi_intel.so.1  000014AA3BD745B6  Unknown               Unknown  Unknown
libmpi_intel.so.1  000014AA3BD74671  Unknown               Unknown  Unknown
libmpi_intel.so.1  000014AA3D628187  Unknown               Unknown  Unknown
libmpi_intel.so.1  000014AA3BDC01FC  PMPI_Allreduce        Unknown  Unknown
libmpifort_intel.  000014AA41C464D6  mpi_allreduce_        Unknown  Unknown
gsi.x              000000000090258F  read_obsmod_mp_re        2030  read_obs.F90
gsi.x              0000000000824D9C  observermod_mp_se         329  observer.F90
gsi.x              0000000000F7D773  glbsoi_                   221  glbsoi.f90
gsi.x              00000000006630EF  gsisub_                   200  gsisub.F90
gsi.x              0000000000411F4D  gsimod_mp_gsimain        2459  gsimod.F90
gsi.x              0000000000411E87  MAIN__                    633  gsimain.f90
```

This is odd.  `gsi.x` should run really fast for a analysis on a 192 x 96 grid.  I've seen `gsi.x` die due to exceeding the wall clock limit when it is built in debug mode.  This shouldn't be the case here.

Line 2030 of `read_obs.F90` is a `mpi_allreduce`
```
!   Sum and distribute number of obs read and used for each input ob group                                                                                                                                         

    call mpi_allreduce(ndata1,ndata,ndat*3,mpi_integer,mpi_sum,mpi_comm_world,&
       ierror)
```

The first run of gdas_anal aborted because
```
+ exglobal_atmos_analysis.sh[500]/bin/ln -sf /lfs/h2/emc/ptmp/emc.global/PR/PR_3498/RUNTESTS/COMROOT/C48mx500_hybAOWCDA_3498/gdas.20210325/00/obs/gdas.t00z.sstvcw.tm00.bufr_d sstviirs
/lfs/h2/emc/ptmp/emc.global/PR/PR_3498/global-workflow/scripts/exglobal_atmos_analysis.sh: line 501: SAILDRONE: unbound variable
+ JGLOBAL_ATMOS_ANALYSIS[92]export err=1
+ JGLOBAL_ATMOS_ANALYSIS[92]err=1
```

Did the rerun wind up on node(s) with slow communications?

ADCollard commented:
Sorry @KateFriedman-NOAA.  I thought this was going to be straightforward!  This should be the same as what we are running in v16.3.25.   I will redo the testing as you ask.

ADCollard committed:
Merge branch 'NOAA-EMC:develop' into develop

ADCollard commented:
I checked out the branch afresh and built and ran the ctests.  I am not seeing any failures so far.

The particular job that failed in the CI ran to completion for me. See `/lfs/h2/emc/stmp/andrew.collard/RUNTESTS/COMROOT/C48mx500_hybAOWCDA/logs/2021032500/gdas_anal.log`.  

```
     ENDING DATE-TIME    JUN 13,2025  17:59:14.898  164  FRI   2460840
     PROGRAM GSI_ANL HAS ENDED.* . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * .
nid003870.dogwood.wcoss2.ncep.noaa.gov 0: *****************RESOURCE STATISTICS*******************************
The total amount of wall time                        = 993.670746
The total amount of time in user mode                = 2323.742173
The total amount of time in sys mode                 = 66.830169
The maximum resident set size (KB)                   = 1565012
Number of page faults without I/O activity           = 527967
Number of page faults with I/O activity              = 1254
Number of times filesystem performed INPUT           = 2359698
Number of times filesystem performed OUTPUT          = 1484888
Number of Voluntary Context Switches                 = 11301
Number of InVoluntary Context Switches               = 3535
*****************END OF RESOURCE STATISTICS*************************
```

Is it possible there was an external issue that caused the failure as @RussTreadon-NOAA suggested?

ADCollard commented:
I do notice that `C48mx500_3DVarAOWCDA` is still running after 37 minutes though and seems to be stuck in the read routines.   I'll report back when it finishes.

ADCollard commented:
It's still running, but it appears that `C48mx500_3DVarAOWCDA` has indeed hung.    I am checking out develop to provide a baseline then going backtrack through the latest GSI commits to see where the ctests start to  misbehave.

CatherineThomas-NOAA commented:
As an extra data point, I ran a series of standalone GSI tests on Dogwood on 2025/6/11 with hash [cdd66d0](https://github.com/NOAA-EMC/GSI/commit/cdd66d0a50330d713fb4630c9d2976f795d4804d) (develop).  I ran at a much higher resolution (C1152 or C768) but I ran ~35 tests and experienced no hangs.  These tests did not include the extra 16.3.25 changes of saildrone, etc.  Just the update of the GSI executable.

ADCollard commented:
GSI is stalling for `C48mx500_hybAOWCDA` when using a GSI branch older than all the saildrone changes but after the 
/lfs/h2/emc/stmp/andrew.collard/RUNTESTS/COMROOT/C48mx500_hybAOWCDA__1ac526f9/logs/2021032500/gdas_anal.log

```
git log
commit 75b54e30adeac7638e28a65b7b7f9355e7679da4 (HEAD)
Author: dkokron <dkokron@users.noreply.github.com>
Date:   Wed May 14 08:10:25 2025 -0500

    Optimize global gsi for GFS v17 (#875)
```

So it doesn't appear to be the obs changes.   But I should re-run with an older `exglobal_atmos_analysis.sh` script to eliminate that too.

RussTreadon-NOAA commented:
@ADCollard , if it turns out that `gsi.x` hangs are associated with [75b54e30](https://github.com/NOAA-EMC/GSI/commit/75b54e30adeac7638e28a65b7b7f9355e7679da4) (GSI PR [#875](https://github.com/NOAA-EMC/GSI/pull/875)), we can ask @dkokron to take a look.

dkokron commented:
Please show me how to run the C48mx500_hybAOWCDA test on dogwood.

ADCollard commented:
I just finished running using the GSI commit before [75b54e30](https://github.com/NOAA-EMC/GSI/commit/75b54e30adeac7638e28a65b7b7f9355e7679da4) and the `gdas_anal` jobs ran to completion.  Unfortunately, the output from my run with  [75b54e30](https://github.com/NOAA-EMC/GSI/commit/75b54e30adeac7638e28a65b7b7f9355e7679da4) was scrubbed over the weekend, so I am going to re-run.

@dkokron If you want to take a look at this case, the first step is to checkout and build [ADCollard:develop](https://github.com/ADCollard/global-workflow.git) and then use the new [generate workflow capability](https://docs.google.com/presentation/d/1azLQfbaVmWojkyUB0n73hkbQbN07LKF4/edit?slide=id.p5#slide=id.p5).   I can guide you through this in more detail if you are not familiar.

dkokron commented:
@ADCollard 
mkdir GFSv163215
cd GFSv163215
git clone --recursive https://github.com/ADCollard/global-workflow.git
cd global-workflow/sorc

Now, which build script and command line args do I use?

ADCollard commented:
In `sorc/`:
```
./build_all.sh
./link_workflow.sh
```

cd to `dev/workflow`

```
./generate_workflows.sh -A GFS-DEV -c -G -t test /lfs/h2/emc/stmp/$USER/RUNTESTS_test
```

This will run the test cases in `dev/ci/cases/pr`

dkokron commented:
I am not in the 'emc' group.  What steps are needed to run as myself?

ADCollard commented:
> I am not in the 'emc' group. What steps are needed to run as myself?

Is this just an issue with the output path?  If so replace `/lfs/h2/emc/stmp/$USER/RUNTESTS_test` with a more suitable path for you.

dkokron commented:
I get the following after setting

`export HOMEDIR=/lfs/h2/hpc/support/daniel.kokron`

```
./generate_workflows.sh -A SUPPORT-HPC -c -G -t test /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/RUNTESTS_test
The RUNTESTS directory /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/RUNTESTS_test already exists.
Would you like to remove /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/RUNTESTS_test?
yes
Running all GFS cases in /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/global-workflow/dev/ci/cases/pr

Running create_experiment.py for 10 cases
C48_ATM
The create_experiment command (./create_experiment.py -y /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/global-workflow/dev/ci/cases/pr/C48_ATM.yaml --overwrite) failed with a non-zero status.  Output:
2025-06-16 18:57:37,298 - INFO     - root        : BEGIN: __main__.input_args
2025-06-16 18:57:37,302 - INFO     - root        :   END: __main__.input_args
2025-06-16 18:57:37,314 - INFO     - root        : Call: setup_expt.main()
2025-06-16 18:57:37,314 - INFO     - setup_expt  : BEGIN: setup_expt.main.main
2025-06-16 18:57:37,352 - WARNING  - setup_expt  : User configuration file does not exist: /u/daniel.kokron/.gwrc, use repository defaults
2025-06-16 18:57:37,356 - INFO     - setup_expt  : Updating host info: ACCOUNT = SUPPORT-HPC
2025-06-16 18:57:37,452 - INFO     - setup_expt  : Jinja2 template found: config.atmensanl.j2
2025-06-16 18:57:37,499 - INFO     - setup_expt  : Jinja2 template found: config.snowanl.j2
2025-06-16 18:57:37,514 - INFO     - setup_expt  : Jinja2 template found: config.base.j2
2025-06-16 18:57:37,631 - INFO     - setup_expt  : Jinja2 template found: config.esnowanl.j2
2025-06-16 18:57:37,635 - INFO     - setup_expt  : Jinja2 template found: config.stage_ic.j2
2025-06-16 18:57:37,640 - INFO     - setup_expt  : Jinja2 template found: config.aeroanl.j2
2025-06-16 18:57:37,700 - INFO     - setup_expt  : Jinja2 template found: config.prepoceanobs.j2
2025-06-16 18:57:37,718 - INFO     - setup_expt  : Jinja2 template found: config.aero.j2
2025-06-16 18:57:37,772 - INFO     - setup_expt  : Jinja2 template found: config.marineanl.j2
2025-06-16 18:57:37,793 - INFO     - setup_expt  : Jinja2 template found: config.atmanl.j2
2025-06-16 18:57:37,804 - INFO     - setup_expt  : Jinja2 template found: config.ocn.j2
2025-06-16 18:57:37,808 - INFO     - setup_expt  : ****************************************************************************************************
2025-06-16 18:57:37,808 - INFO     - setup_expt  : EXPDIR: /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/RUNTESTS_test/EXPDIR/C48_ATM_test
2025-06-16 18:57:37,808 - INFO     - setup_expt  : ROTDIR: /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/RUNTESTS_test/COMROOT/C48_ATM_test
2025-06-16 18:57:37,808 - INFO     - setup_expt  : ****************************************************************************************************
2025-06-16 18:57:37,808 - INFO     - setup_expt  :   END: setup_expt.main.main
2025-06-16 18:57:37,808 - INFO     - root        : Call: setup_xml.main()
2025-06-16 18:57:37,808 - INFO     - setup_xml   : BEGIN: setup_xml.main.main
Traceback (most recent call last):
  File "./create_experiment.py", line 116, in <module>
    setup_xml.main(setup_xml_args)
  File "/lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/global-workflow/dev/workflow/wxflow/logger.py", line 266, in wrapper
    retval = func(*args, **kwargs)
  File "/lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/global-workflow/dev/workflow/setup_xml.py", line 103, in main
    check_dir_writable(base[dk])
  File "/lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/global-workflow/dev/workflow/wxflow/attrdict.py", line 84, in __missing__
    raise KeyError(name)
KeyError: 'HOMEDIR'
```

ADCollard commented:
> I get the following after setting
> 
> `export HOMEDIR=/lfs/h2/hpc/support/daniel.kokron`
> 
> ```
> ./generate_workflows.sh -A SUPPORT-HPC -c -G -t test /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/RUNTESTS_test
> The RUNTESTS directory /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/RUNTESTS_test already exists.
> Would you like to remove /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/RUNTESTS_test?
> yes
> Running all GFS cases in /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/global-workflow/dev/ci/cases/pr
> 
> Running create_experiment.py for 10 cases
> C48_ATM
> The create_experiment command (./create_experiment.py -y /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/global-workflow/dev/ci/cases/pr/C48_ATM.yaml --overwrite) failed with a non-zero status.  Output:
> 2025-06-16 18:57:37,298 - INFO     - root        : BEGIN: __main__.input_args
> 2025-06-16 18:57:37,302 - INFO     - root        :   END: __main__.input_args
> 2025-06-16 18:57:37,314 - INFO     - root        : Call: setup_expt.main()
> 2025-06-16 18:57:37,314 - INFO     - setup_expt  : BEGIN: setup_expt.main.main
> 2025-06-16 18:57:37,352 - WARNING  - setup_expt  : User configuration file does not exist: /u/daniel.kokron/.gwrc, use repository defaults
> 2025-06-16 18:57:37,356 - INFO     - setup_expt  : Updating host info: ACCOUNT = SUPPORT-HPC
> 2025-06-16 18:57:37,452 - INFO     - setup_expt  : Jinja2 template found: config.atmensanl.j2
> 2025-06-16 18:57:37,499 - INFO     - setup_expt  : Jinja2 template found: config.snowanl.j2
> 2025-06-16 18:57:37,514 - INFO     - setup_expt  : Jinja2 template found: config.base.j2
> 2025-06-16 18:57:37,631 - INFO     - setup_expt  : Jinja2 template found: config.esnowanl.j2
> 2025-06-16 18:57:37,635 - INFO     - setup_expt  : Jinja2 template found: config.stage_ic.j2
> 2025-06-16 18:57:37,640 - INFO     - setup_expt  : Jinja2 template found: config.aeroanl.j2
> 2025-06-16 18:57:37,700 - INFO     - setup_expt  : Jinja2 template found: config.prepoceanobs.j2
> 2025-06-16 18:57:37,718 - INFO     - setup_expt  : Jinja2 template found: config.aero.j2
> 2025-06-16 18:57:37,772 - INFO     - setup_expt  : Jinja2 template found: config.marineanl.j2
> 2025-06-16 18:57:37,793 - INFO     - setup_expt  : Jinja2 template found: config.atmanl.j2
> 2025-06-16 18:57:37,804 - INFO     - setup_expt  : Jinja2 template found: config.ocn.j2
> 2025-06-16 18:57:37,808 - INFO     - setup_expt  : ****************************************************************************************************
> 2025-06-16 18:57:37,808 - INFO     - setup_expt  : EXPDIR: /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/RUNTESTS_test/EXPDIR/C48_ATM_test
> 2025-06-16 18:57:37,808 - INFO     - setup_expt  : ROTDIR: /lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/RUNTESTS_test/COMROOT/C48_ATM_test
> 2025-06-16 18:57:37,808 - INFO     - setup_expt  : ****************************************************************************************************
> 2025-06-16 18:57:37,808 - INFO     - setup_expt  :   END: setup_expt.main.main
> 2025-06-16 18:57:37,808 - INFO     - root        : Call: setup_xml.main()
> 2025-06-16 18:57:37,808 - INFO     - setup_xml   : BEGIN: setup_xml.main.main
> Traceback (most recent call last):
>   File "./create_experiment.py", line 116, in <module>
>     setup_xml.main(setup_xml_args)
>   File "/lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/global-workflow/dev/workflow/wxflow/logger.py", line 266, in wrapper
>     retval = func(*args, **kwargs)
>   File "/lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/global-workflow/dev/workflow/setup_xml.py", line 103, in main
>     check_dir_writable(base[dk])
>   File "/lfs/h2/hpc/support/daniel.kokron/Projects/GSI/GFSv163215/global-workflow/dev/workflow/wxflow/attrdict.py", line 84, in __missing__
>     raise KeyError(name)
> KeyError: 'HOMEDIR'
> ```

I am not sure, but I had a similar error when I had `PTMP` set.   Does this work if you `unset HOMEDIR`?

CoryMartin-NOAA commented:
Because of the way the workflow works, you can't `export` those variables, they need to be set inline or in your ~/.gwrc file

dkokron commented:
Okay, I copied the template from dev/parm/workflow/gwrc to /u/daniel.kokron/.gwrc and made some changes in that file.  I also unset HOMEDIR.  Now the I get the following.

```
Running create_experiment.py for 10 cases
C48_ATM
C48mx500_3DVarAOWCDA
C48mx500_hybAOWCDA
C48_S2SW_extended
C48_S2SW
C96_atm3DVar_extended
C96C48_hybatmDA
C96C48_hybatmsnowDA
C96C48mx500_S2SW_cyc_gfs
C96C48_ufs_hybatmDA

Updating the existing crontab

Success!!
```
There are 11 jobs queued or running.

RussTreadon-NOAA commented:
**WCOSS2 test**

Did the following on Dogwood
1. copy C48mx500_3DVarAOWCDA run directory for 20210325 00Z gdas_anal to `/lfs/h2/emc/da/noscrub/russ.treadon/CASES/GSI_HANG/anal.3355900`
2. create script `rungsi.sh` to run `gsi.x` using files in the above copied run directory.  Use `gsi.x` built from the authoritative NOAA-EMC/GSI `develop` at [955580784](https://github.com/NOAA-EMC/GSI/commit/955580784145fa81dd9b3b9056626044dc13b37a).  This is one commit behind the current head of GSI `develop`.  It's six commits after the GFS v17 optimizations.
3. `gsi.x` configured to run 84 tasks with 25 tasks per node and 5 threads per task.   This is how `gsi.x` is configured in C48mx500_3DVarAOWCDA
4. submit script.  `gsi.x` hung.  job killed when it reached specified 20 minute wall clock.  log file is `gsi.o46058945`.
5. modify `rungsi.sh` to run `gsi.x` using 120 tasks with 40 tasks per node and 3 threads per task
6. submit script.  `gsi.x` ran to completion in  879.719361 seconds. log file is `gsi.o46061209`.

@dkokron :  The last `gsi.x` printout from the hung run is
```
 READ_SATWND: ictype(nc),rmesh,pflag,nlevp,pmesh,nc uv                254    200.00    1   12    100.00  139    0      2.00
 READ_SATWND: ictype(nc),rmesh,pflag,nlevp,pmesh,nc uv                254    200.00    1   12    100.00  140    0      2.00
 READ_SATWND,nread,ndata,nreal,nodata=     3070150     1312856          27
     2625712
```
Are we dealing with issues in the data readers which were optimized as part of GSI PR [#875](https://github.com/NOAA-EMC/GSI/pull/875)?
 

To run this case and figure out why the code hangs with the 84 task configuration, copy the following to your space
- directory `/lfs/h2/emc/da/noscrub/russ.treadon/CASES/GSI_HANG/anal.3355900`
- script: `/lfs/h2/emc/da/noscrub/russ.treadon/CASES/GSI_HANG/rungsi.sh`
 
Update `rungsi.sh` to use your copy of the run directory.  Don't assume I'll keep my copy.  I'll eventually forget why I copied it in the first place and delete it.

dkokron commented:
80 of the ranks are sitting in a MPI_Allreduce at line 2016 of /lfs/h2/emc/da/noscrub/andrew.collard/git/global-workflow_ADC_test/sorc/gsi_enkf.fd/src/gsi/read_obs.F90 while the other 4 are sitting in read_bufrtovs().

Here is the stack trace following line 1743
```
#23 0x000014fa260464d6 in pmpi_allreduce__ () from /opt/cray/pe/lib64/libmpifort_intel.so.12
#24 0x0000000000cedaa9 in combine_radobs (mype_sub=<error reading variable: Cannot access memory at address 0x1>, mype_root=11253572, npe_sub=167972, mpi_comm_sub=<error reading variable: Cannot access memory at address 0x1019>, nele=<error reading variable: Cannot access memory at address 0x0>, itxmax=<error reading variable: Cannot access memory at address 0x0>, nread=1457150, ndata=0, data_all=<error reading variable: value requires 9031376672395495424 bytes, which is more than max-value-size>, data_crit=<error reading variable: value requires 1125890864574336 bytes, which is more than max-value-size>, nrec=<error reading variable: value requires 562945432287168 bytes, which is more than max-value-size>) at /lfs/h2/emc/da/noscrub/andrew.collard/git/global-workflow_ADC_test/sorc/gsi_enkf.fd/src/gsi/combine_radobs.f90:70
#25 0x0000000001273238 in read_bufrtovs (mype=<error reading variable: Cannot access memory at address 0x1>, val_tovs=0, ithin=1, isfcalc=1, rmesh=0, jsatid=<error reading variable: value requires 140736355961440 bytes, which is more than max-value-size>, gstime=22736160, infile=<error reading variable: value requires 140736355961440 bytes, which is more than max-value-size>, lunout=81, obstype=<error reading variable: value requires 140736355961440 bytes, which is more than max-value-size>, nread=1457150, ndata=0, nodata=0, twind=3, sis=..., mype_root=0, mype_sub=0, npe_sub=2, mpi_comm_sub=-2080374783, nobs=<error reading variable: value requires 562945423845760 bytes, which is more than max-value-size>, nrec_start=697, nrec_start_ears=999999, nrec_start_db=999999, dval_use=.FALSE., radmod=..., .tmp.JSATID.len_V$925=11, .tmp.INFILE.len_V$936=120, .tmp.OBSTYPE.len_V$947=10, .tmp.SIS.len_V$95b=20) at /lfs/h2/emc/da/noscrub/andrew.collard/git/global-workflow_ADC_test/sorc/gsi_enkf.fd/src/gsi/read_bufrtovs.f90:1271
#26 0x000000000090a2ff in read_obsmod::read_obs (ndata=..., mype=11253572) at /lfs/h2/emc/da/noscrub/andrew.collard/git/global-workflow_ADC_test/sorc/gsi_enkf.fd/src/gsi/read_obs.F90:1743
```
Three of the four sitting under read_bufrtovs() are in one of my new OpenMP directives at line 930 of 
/lfs/h2/emc/da/noscrub/andrew.collard/git/global-workflow_ADC_test/sorc/gsi_enkf.fd/src/gsi/read_bufrtovs.f90.  The actual hang appears to be in the call deter_sfc_fov() inside that loop.

The other rank is sitting at an MPI_Allreduce() (on a sub communicator of comm_world) at read_bufrtovs (line 1271) -> combine_radobs (line 70).  More to follow.

RussTreadon-NOAA commented:
**C48mx500_3DVarAOWCDA hang**

Another data point:  build GSI `develop` in debug mode.  Run C48mx500_3DVarAOWCDA case with 85 tasks.  Execution is slow but makes it into minimization before wall clock terminates job.   Recompile `develop` in optimized mode.  Code hangs as described above.

GSI issue [#898](https://github.com/NOAA-EMC/GSI/issues/898) has been created to document this problem and its resolution.

@dkokron and others:  we can use GSI issue [#898](https://github.com/NOAA-EMC/GSI/issues/898) to discuss the `gsi.x` hang and how to resolve this problem.

RussTreadon-NOAA commented:
GSI PR [#899](https://github.com/NOAA-EMC/GSI/pull/899) was merged into GSI `develop` at [`964bcc3`](https://github.com/NOAA-EMC/GSI/commit/964bcc37597310d99d22627d769d5e5a82d4dd5a).  This PR contains a fix which allows the failed g-w CI C48mx500_hybAOWCDA  gdas_anal to run to completion.    

We should update the `sorc/gsi_enkf.fd` hash in [ADCollard:develop](https://github.com/ADCollard/global-workflow/tree/develop) to [`964bcc3`](https://github.com/NOAA-EMC/GSI/commit/964bcc37597310d99d22627d769d5e5a82d4dd5a).

ADCollard commented:
> GSI PR [#899](https://github.com/NOAA-EMC/GSI/pull/899) was merged into GSI `develop` at [`964bcc3`](https://github.com/NOAA-EMC/GSI/commit/964bcc37597310d99d22627d769d5e5a82d4dd5a). This PR contains a fix which allows the failed g-w CI C48mx500_hybAOWCDA gdas_anal to run to completion.
> 
> We should update the `sorc/gsi_enkf.fd` hash in [ADCollard:develop](https://github.com/ADCollard/global-workflow/tree/develop) to [`964bcc3`](https://github.com/NOAA-EMC/GSI/commit/964bcc37597310d99d22627d769d5e5a82d4dd5a).

Doing it now!

I will open a fresh clone of `global-workflow` and re-run the CI jobs before re-opening the pull request.

ADCollard committed:
Merge branch 'NOAA-EMC:develop' into develop

ADCollard committed:
Checkout GSI with bugfix

ADCollard commented:
So it took a little longer than expected (apparently build_compute.sh does not build everything by default!) but the CI tests now build for me.  Thanks @dkokron and @RussTreadon-NOAA for finding the (very old) bug!

I have since identified an issue that needs to be addressed with the Windborne obs error vertical interpolation in GSI.   Should we wait for that update or add that in a separate PR.

DavidHuber-NOAA reviewed (approved):
Looks good.

DavidHuber-NOAA committed:
Merge branch 'develop' into develop

DavidHuber-NOAA commented:
Launching CI on Hera.  All systems should be tested, but those will be triggered after CI passes on Hera.

aerorahul committed:
Merge branch 'develop' into develop

KateFriedman-NOAA reviewed (approved):
Thanks @ADCollard !

KateFriedman-NOAA committed:
Merge branch 'develop' into develop

KateFriedman-NOAA committed:
Merge branch 'develop' into develop

KateFriedman-NOAA commented:
Removed Orion and Hercules ready labels. MSU is down for monthly maintenance today.

KateFriedman-NOAA commented:
CI Tests set up to run in /lfs/h2/emc/ptmp/emc.global/PR/PR_3498/RUNTESTS on WCOSS

KateFriedman-NOAA commented:
CI passed on WCOSS2:
```
Thu Jun 26 13:52:37 UTC 2025
******** C48_ATM_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jun 23 2025 17:20:05    Jun 23 2025 18:21:49

******** C48mx500_3DVarAOWCDA_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jun 23 2025 17:20:07    Jun 23 2025 17:32:14
202103250000        Done    Jun 23 2025 17:20:07    Jun 23 2025 19:52:00

******** C48mx500_hybAOWCDA_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jun 23 2025 17:20:09    Jun 23 2025 17:32:22
202103250000        Done    Jun 23 2025 17:20:09    Jun 23 2025 18:37:10

******** C48_S2SW_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jun 23 2025 17:20:11    Jun 23 2025 18:32:07

******** C48_S2SWA_gefs_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jun 23 2025 17:20:14    Jun 23 2025 18:27:32

******** C48_S2SW_extended_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jun 23 2025 17:20:17    Jun 23 2025 20:12:10
202103231800        Done    Jun 23 2025 17:20:17    Jun 23 2025 20:22:21

******** C96_atm3DVar_extended_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jun 23 2025 17:20:20    Jun 23 2025 17:37:39
202112210000        Done    Jun 23 2025 17:20:20    Jun 23 2025 21:47:24
202112210600        Done    Jun 23 2025 17:20:20    Jun 23 2025 22:47:15
202112211200        Done    Jun 23 2025 17:42:38    Jun 23 2025 23:27:16
202112211800        Done    Jun 23 2025 21:52:37    Jun 24 2025 02:42:15

******** C96C48_hybatmDA_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jun 23 2025 17:20:22    Jun 23 2025 17:37:46
202112210000        Done    Jun 23 2025 17:20:22    Jun 23 2025 20:27:32
202112210600        Done    Jun 23 2025 17:20:22    Jun 23 2025 20:07:37

******** C96C48_hybatmsnowDA_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jun 23 2025 17:20:25    Jun 23 2025 17:37:53
202112201800        Done    Jun 23 2025 17:20:25    Jun 23 2025 20:22:44
202112210000        Done    Jun 23 2025 17:20:25    Jun 23 2025 20:07:43

******** C96C48mx500_S2SW_cyc_gfs_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jun 23 2025 17:20:27    Jun 23 2025 17:38:00
202112201800        Done    Jun 23 2025 17:20:27    Jun 23 2025 20:27:44
202112210000        Done    Jun 23 2025 17:20:27    Jun 23 2025 21:03:08
202112211800        Done    Jun 23 2025 17:43:00    Jun 23 2025 20:57:52

******** C96C48_ufs_hybatmDA_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202402231800        Done    Jun 23 2025 17:20:29    Jun 23 2025 17:38:07
202402240000        Done    Jun 23 2025 17:20:29    Jun 23 2025 20:33:01
202402240600        Done    Jun 23 2025 17:20:29    Jun 24 2025 15:38:09

******** C96_gcafs_cycled_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jun 23 2025 17:20:31    Jun 23 2025 17:38:14
202112201800        Done    Jun 23 2025 17:20:31    Jun 23 2025 20:18:12
202112210000        Done    Jun 23 2025 17:20:31    Jun 23 2025 20:03:15

******** C96mx100_S2S_3823 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
199405010000        Done    Jun 23 2025 17:20:34    Jun 23 2025 18:53:31
```

CatherineThomas-NOAA commented:
The S2S failure on Orion is showing an HDF error:
```
30: FATAL from PE     0: NetCDF: HDF error: netcdf_file_open:INPUT/MOM.res.nc
```
The `MOM.res.nc` file appears to be the right size (not different than the other files):
```
orion-login-4[18] cthomas$ ls -l /work2/noaa/global/role-global/stmp/ORION/RUNDIRS/C96mx100_S2S_92fcac6c/sfs.1994050100/sfsfcst.1994050100/restart/MOM6_RESTART/
total 4478540
-rw-r----- 1 role-global noaa-hpc 917198614 Jun 26 08:15 19940501.120000.MOM.res.nc
-rw-r----- 1 role-global noaa-hpc 917198614 Jun 26 08:21 19940502.000000.MOM.res.nc
-rw-r----- 1 role-global noaa-hpc 917198614 Jun 26 08:27 19940502.120000.MOM.res.nc
-rw-r----- 1 role-global noaa-hpc 917198614 Jun 26 08:34 19940503.000000.MOM.res.nc
-rw-r----- 1 role-global noaa-hpc 917198614 Jun 26 08:53 19940503.120000.MOM.res.nc
```
I don't see how anything in this PR could affect the ocean files. Just a machine hiccup?

KateFriedman-NOAA commented:
> I don't see how anything in this PR could affect the ocean files. Just a machine hiccup?

Perhaps. I have booted the job to see if it runs now.

KateFriedman-NOAA commented:
> Perhaps. I have booted the job to see if it runs now.

It dies on Orion again because of a date check with the restarts (dies before prior failure point):
```
+ forecast_det.sh[114]RERUN=YES
+ forecast_det.sh[115]RERUN_DATE=1994050412
+ forecast_det.sh[118][[ 1994050412 -ge 1994050300 ]]
+ forecast_det.sh[119]echo 'FATAL ERROR Warm start detected, but restart date (1994050412) is at/after model end date (1994050300)'
FATAL ERROR Warm start detected, but restart date (1994050412) is at/after model end date (1994050300)
+ forecast_det.sh[120]exit 1
```
I see that the `RERUN_DATE` is different in the third attempt (1994050412) I just booted:
```
[role-global@orion-login-3 C96mx100_S2S_92fcac6c]$ grep RERUN_DATE= logs/1994050100/sfs_fcst_mem000_seg0.log*
logs/1994050100/sfs_fcst_mem000_seg0.log:+ forecast_det.sh[115]RERUN_DATE=1994050412
logs/1994050100/sfs_fcst_mem000_seg0.log.0:+ forecast_det.sh[115]RERUN_DATE=1994050212
```
That date comes from `rdate`:
```
logs/1994050100/sfs_fcst_mem000_seg0.log:+ forecast_det.sh[62]rdate=1994050412
logs/1994050100/sfs_fcst_mem000_seg0.log.0:+ forecast_det.sh[62]rdate=1994050212
```
...which comes from `filename`:
```
[role-global@orion-login-3 C96mx100_S2S_92fcac6c]$ grep filename= logs/1994050100/sfs_fcst_mem000_seg0.log* | grep forecast_det
logs/1994050100/sfs_fcst_mem000_seg0.log:+ forecast_det.sh[61]filename=19940504.120000.coupler.res
logs/1994050100/sfs_fcst_mem000_seg0.log.0:+ forecast_det.sh[61]filename=19940502.120000.coupler.res
```
This looks like an issue with the restart calculation perhaps. @aerorahul can you take a look?

KateFriedman-NOAA commented:
> /work2/noaa/global/role-global/GFS_CI_CD/ORION/CI/3498/RUNTESTS/COMROOT/C96C48mx500_S2SW_cyc_gfs_92fcac6c/logs/2021122018/gfs_marineanlfinal.log
> /work2/noaa/global/role-global/GFS_CI_CD/HERCULES/CI/3498/RUNTESTS/COMROOT/C48mx500_hybAOWCDA_92fcac6c/logs/2021032500/gdas_marineanlfinal.log

Both of these jobs (Orion and Hercules) die due to a missing input file:
```
^[[38;5;196m2025-06-26 09:17:23,659 - ERROR    - file_utils  : Source file '/work2/noaa/global/role-global/stmp/HERCULES/RUNDIRS/C48mx500_hybAOWCDA_92fcac6c/gdas.2021032500/gdasmarineanalysis.2021032500/marinevariational/ocn.inc.nc' does not exist and is required, ABORT!^[[0m
NoneType: None
^[[38;5;196m2025-06-26 09:17:23,659 - ERROR    - file_utils  : Source file '/work2/noaa/global/role-global/stmp/HERCULES/RUNDIRS/C48mx500_hybAOWCDA_92fcac6c/gdas.2021032500/gdasmarineanalysis.2021032500/marinevariational/ocn.inc.nc' does not exist and is required, ABORT!^[[0m
NoneType: None
Traceback (most recent call last):
  File "/work2/noaa/global/role-global/GFS_CI_CD/HERCULES/CI/3498/global-workflow/scripts/exglobal_marine_analysis_finalize.py", line 24, in <module>
    MarineAnl.finalize()
  File "/work2/noaa/global/role-global/GFS_CI_CD/HERCULES/CI/3498/global-workflow/ush/python/wxflow/logger.py", line 266, in wrapper
    retval = func(*args, **kwargs)
  File "/work2/noaa/global/role-global/GFS_CI_CD/HERCULES/CI/3498/global-workflow/ush/python/pygfs/task/marine_analysis.py", line 199, in finalize
    FileHandler(soca_finalize_list).sync()
  File "/work2/noaa/global/role-global/GFS_CI_CD/HERCULES/CI/3498/global-workflow/ush/python/wxflow/file_utils.py", line 65, in sync
    sync_factory[action](files)
  File "/work2/noaa/global/role-global/GFS_CI_CD/HERCULES/CI/3498/global-workflow/ush/python/wxflow/file_utils.py", line 69, in copy_req
    FileHandler._copy_files(filelist, required=True)
  File "/work2/noaa/global/role-global/GFS_CI_CD/HERCULES/CI/3498/global-workflow/ush/python/wxflow/file_utils.py", line 105, in _copy_files
    raise FileNotFoundError(f"Source file '{src}' does not exist")
FileNotFoundError: Source file '/work2/noaa/global/role-global/stmp/HERCULES/RUNDIRS/C48mx500_hybAOWCDA_92fcac6c/gdas.2021032500/gdasmarineanalysis.2021032500/marinevariational/ocn.inc.nc' does not exist
```
This appears to be a file that is produced during the `marineanlchkpt` job (as seen in the CI test on WCOSS2):
```
WCOSS2 (BACKUPSYS-D) C48mx500_hybAOWCDA_3498> grep ocn.inc.nc logs/*/*
logs/2021032500/gdas_marineanlchkpt.log:nid001055.dogwood.wcoss2.ncep.noaa.gov 0: domain: ocn rename: /lfs/h2/emc/stmp/emc.global/RUNDIRS/C48mx500_hybAOWCDA_3498/gdas.2021
032500/gdasmarineanalysis.2021032500/marinevariational/ocn.mom6_iau.incr.2021-03-24T21:00:00Z.nc to .//ocn.inc.nc
logs/2021032500/gdas_marineanlfinal.log:2025-06-25 13:57:58,334 - INFO     - file_utils  : Copied /lfs/h2/emc/stmp/emc.global/RUNDIRS/C48mx500_hybAOWCDA_3498/gdas.20210325
00/gdasmarineanalysis.2021032500/marinevariational/ocn.inc.nc to /lfs/h2/emc/ptmp/emc.global/PR/PR_3498/RUNTESTS/COMROOT/C48mx500_hybAOWCDA_3498/gdas.20210325/00//analysis/ocean/gdas.t00z.ocninc.nc
```
Looking at the Orion failure, the gdas version of this job ran fine but the gfs version failed. Who can look at these jobs?

KateFriedman-NOAA commented:
Seeing other random failures now with the Orion and Hercules CI tests. Wondering if these are machine issues now.

KateFriedman-NOAA commented:
Since it appears that Orion/Hercules may be dealing with disk issues, we are going to ignore the CI testing results on those platforms. The CI tests on Gaea C6 are wrapping up now. Once it reports success, we'll consider this PR good to merge.

KateFriedman-NOAA committed:
Merge branch 'develop' into develop

KateFriedman-NOAA committed:
Merge branch 'develop' into develop
